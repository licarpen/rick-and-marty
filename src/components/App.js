/* eslint-disable react/jsx-no-comment-textnodes */
/* eslint-disable react/jsx-key */
import React from 'react';
import Header from './Header.js';
import Footer from './Footer.js';
import Character from './Character.js';
import Chance from 'chance';
const chance = new Chance();
import styles from './Character.css';

const characters = [...Array(8)]
  // eslint-disable-next-line no-unused-vars
  .map(i => ({ name: chance.name(), image: 'http://placegoat.com/300/300', status: chance.pickone(['alive', 'dead', 'unknown']), species: chance.animal(), gender: chance.pickone(['male', 'female', 'nonbinary']), origin: chance.city(), lastLocation: chance.city() }));

const App = () => (
  <React.Fragment>
    <Header />
    // eslint-disable-next-line react/jsx-no-comment-textnodes
    <div className={styles.container}>
      // eslint-disable-next-line react/jsx-key
      {characters.map(character => <Character image={character.image} name={character.name} status={character.status} species={character.species} gender={character.gender} origin={character.origin} lastLocation={character.lastLocation}/>)};
    </div>
    <Footer />
  </React.Fragment>
);

export default App;
