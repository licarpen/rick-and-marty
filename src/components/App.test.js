import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('chance', () => (class Chance { 
  name(){return 'my name';}
  city(){return 'my city';}
  pickone(){return 'great choice';}
  animal(){return 'the best animal';}
}));

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
