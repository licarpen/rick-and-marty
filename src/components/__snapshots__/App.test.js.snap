// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Fragment>
  <Header />
  // eslint-disable-next-line react/jsx-no-comment-textnodes
  <div
    className="container"
  >
    // eslint-disable-next-line react/jsx-key
    <Character
      gender="male"
      image="http://placegoat.com/300/300"
      lastLocation="Iwakascaj"
      name="Katie Young"
      origin="Vaurawuc"
      species="Crane Fly"
      status="dead"
    />
    <Character
      gender="female"
      image="http://placegoat.com/300/300"
      lastLocation="Vuvvudu"
      name="Frances Mann"
      origin="Gulpoha"
      species="Cows"
      status="alive"
    />
    <Character
      gender="female"
      image="http://placegoat.com/300/300"
      lastLocation="Acmekwog"
      name="Kenneth Myers"
      origin="Jopunoro"
      species="Aldabra Tortoise"
      status="unknown"
    />
    <Character
      gender="female"
      image="http://placegoat.com/300/300"
      lastLocation="Nadpaepi"
      name="Albert Little"
      origin="Rukemi"
      species="Yelloweye Rockfish"
      status="alive"
    />
    <Character
      gender="male"
      image="http://placegoat.com/300/300"
      lastLocation="Ejegear"
      name="Joseph Long"
      origin="Regubaon"
      species="Ant"
      status="alive"
    />
    <Character
      gender="female"
      image="http://placegoat.com/300/300"
      lastLocation="Ivarapu"
      name="Larry Marsh"
      origin="Lajasu"
      species="Red Panda"
      status="unknown"
    />
    <Character
      gender="nonbinary"
      image="http://placegoat.com/300/300"
      lastLocation="Fecilpa"
      name="Minnie McLaughlin"
      origin="Nimiczaw"
      species="Bustard"
      status="alive"
    />
    <Character
      gender="female"
      image="http://placegoat.com/300/300"
      lastLocation="Evehiku"
      name="Bill Morrison"
      origin="Fakcagu"
      species="Toad"
      status="unknown"
    />
    ;
  </div>
  <Footer />
</Fragment>
`;
